### ISSUE TITLE
Questions on technical details

### ISSUE THREAD

**Issue by Felix-Zhenghao** [Ingrid789/SkillMimic-V2#2]
@wyhuai Sorry to put the questions here because I can't find your email.

I'm an undergrad researcher at NUS and have four questions on the technical details of this paper:

1. In section 4.3 STF the paper writes "unique transition directions must be established for each neighborhood state to ensure convergence". However, I don't see why the proposed similarity-based STF can realize this "unique transition direction". Could you further explain?

2. History encoder. Intuitively, the history encoder is trained such that $(s_t,h_t,c)$ can predict next state $s_{t+1}$. However, this may leak information when using $(s_t,h_t,c)$ as the policy input: $\pi(a_t | s_t,h_t,c)$. Do you have any comments on this potential informaiton leakage? Have you tried to directly use history states (w/o encoding them) to condition the policy? If so, what is the performance?

3. In experiment on ParaHome, you condition the policy on timestep. Since ParaHome is not a very big dataset (486 minutes), condition the policy on timestep may potentially make the policy to cheat by remember action at a specific timestep. Furthermore, action may well not be conditioned on timestep because absolute timestep is intuitively not informative to action (I can do any action at a specific timestep as long as the task progression allow me to do that). So have you done experiments on conditioning the policy simply with $h_t$ and without timestep $c$? (Or $SM+\epsilon-NSI$ does this?) If so, what is the performance?

4. How do you deal with state masks? In the paper you call them "temporal buffers" and says "they can be repaired". Could you provide more details of how they can be "repaired"? In the algorithm 1 of your appendix, I don't see you solution of dealing with masks after they are added.

I really like this paper for its good performance and insightful problem formulation. Thank you if you can clarify my doubts!!

Best,
Zhenghao (Felix) Chi

**Comment by wyhuai** [comment]
@Felix-Zhenghao Hi Felix, thanks for your attention. (1) Consider a state C that exists in the neighborhoods of both state A and state B. When the agent initiates from or is perturbed to state C, the policy faces ambiguity in deciding whether to learn C->A or C->B. Our solution: if C is close to A than B, then we construct a path C->A. (2) history is about the past, not the future, so there is no leakage. No encoding will cause overfitting and is hard to learn(converge). (3) Only SM+T use timesteps, other methods does not. (4) State masks is just like masks in image inpainting tasks. For example, the reference data is A->B->Mask->Mask->C->D, only ABCD will have imitation reward and "Mask" will excluded from reward computation. In this case, the policy will learn (unsupervised) how to act during "Mask" period so that it can catch the following chain C->D. Of course, There exist multiple solution to "inpaint" the "Mask" period, but usually the policy will act similar to learned patterns.

**Comment by Felix-Zhenghao** [comment]
Thank you! I will think over these question based on your answer :)
